File: src/js/main/commands/RichJson_merge_folder.js
/*
	merges the given folder in one object
*/

import {__executeFolderCommand} from "./RichJson_folder.js";
import {getKeysSorted, mergeIntoTarget} from "../RichJsonHelper.js";

export function __executeMergeFolderCommand(parser, context) {
    let folder = __executeFolderCommand(parser, context);
    let entry_names = getKeysSorted(folder);
    let rv = {};

    if (entry_names.length !== 0) {
        for (let i = 0; i < entry_names.length; ++i) {
            mergeIntoTarget(rv, folder[entry_names[i]]);
        }
    }

    return rv;
}

File: src/python/main/commands/rich_json_merge_folder.py
"""
Merges the given folder in one object
"""
from .rich_json_folder import _execute_folder_command
from ..helper.rich_json_helper import get_keys_sorted, merge_into_target


def _execute_merge_folder_command(parser, context):
    folder = _execute_folder_command(parser, context)
    entry_names = get_keys_sorted(folder)
    rv = {}

    if len(entry_names) != 0:
        for name in entry_names:
            merge_into_target(rv, folder[name])

    return rv
